### ISSUE TITLE
Label with background parameter missing some regions.

### ISSUE THREAD

**Issue by eduardohenriquearnold** [scikit-image/scikit-image#941]
Hello there. First, I would like to thank you for the great work done in this package, it is very useful and easy to use.

I am having problem detecting characters in an image and I am wondering if it is not a problem with the label function.

First, I load the image and apply a threshold function, so that it's pixels level are either 0 or 1. Then I apply the skimage.morphology.label() function with the argument background=1, as it is supposed to ignore the white background and the white objects inside letters.

The only problem is that some letters are not detected. In the attached image it is shown the image result of the label function. You can clearly see that all the other characters have been successfully identified, but not 'f', as it doesn't get a proper label value. Is there any reason for this to be happening?

Result image:
![asd](https://f.cloud.github.com/assets/4097245/2453367/18a99622-aedd-11e3-8876-3aac3f37148e.png)

Original Image:
![training1](https://f.cloud.github.com/assets/4097245/2453385/64322fdc-aedd-11e3-910f-d40afc603d69.png)

This is the code.
```
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches

from scipy.misc import imread,imresize

from skimage.morphology import label
from skimage.measure import regionprops
image = imread(path ,1)

#apply threshold in order to make the image binary
bw = image > 120

# label image regions
label_image = label(bw,neighbors=8, background=1)
fig, ax = plt.subplots(ncols=1, nrows=1, figsize=(6, 6))
ax.imshow(label_image, cmap='gray', vmin=-1, vmax=1)

for region in regionprops(label_image, ['Area', 'BoundingBox']):
    # skip small images
    if region['Area'] > 0:
    
        # draw rectangle around segmented coins
        minr, minc, maxr, maxc = region['BoundingBox']
        rect = mpatches.Rectangle((minc, minr), maxc - minc, maxr - minr,
                              fill=False, edgecolor='red', linewidth=2)
        ax.add_patch(rect)
        

plt.show()
```

**Comment by emmanuelle** [comment]
TODO.txt mentions for 0.12: "Change `label` to mark background as 0, not -1, which is consistent with
  SciPy's labelling.". @scikit-image/core do we still plan to do it before the release? Anybody who wants to take care of a PR about this point?

**Comment by jni** [comment]
@emmanuelle I should probably take this on since I find it so infuriating. =P

**Comment by emmanuelle** [comment]
@jni  then I'll review your PR because I think the same way!

**Comment by emmanuelle** [comment]
@jni @stefanv @ahojnnes @JDWarner @tonysyu I've taken over a branch started by @stefanv to change the value of the background label. Now I have a question for you: with
```
x = np.array([[0, 1],
              [1, 0]], dtype=int)
```
do we want
```
>>> measure.label(x, 4)
array([[0, 1],
       [2, 3]])
```
or 
```
>>> measure.label(x, 4)
array([[1, 2],
       [3, 4]])
```
or 
```
>>> measure.label(x, 4)
array([[0, 1],
       [2, 0]])
```
??

That is, do we want that ``background=0`` is the default behavior (last result), or that by default, there is no background? In the latter case, should labels start at 0 or 1?

**Comment by emmanuelle** [comment]
@scikit-image/core modifying the value of background pixels might the last important step (in my opinion) before the release, therefore any comment / advice is very much appreciated!

**Comment by JDWarner** [comment]
To me, the most intuitive behavior is the last one. Document an explicit background value (assumed to be zero by default) with labels starting at 1.

**Comment by zpincus** [comment]
Compatibility with the conventions of `scipy.ndimage` (which also has background as 0) is a good thing IMO. This convention is a little awkward in cases where you need to construct the indices of all the found objects (`object_indices = np.arange(1, num_objects+1)`), but other than that, it works well.

Not having a background at all might be useful in certain cases, but I suspect will be be really confusing if it's the default. (See e.g. the example at the start of this issue, with *tons* of non-connected but still clearly background regions.) If -1 was tricky for people to deal with, no background at all will be really tough.

So I agree with Josh: use zero as the explicit background value with labels starting at 1.

**Comment by emmanuelle** [comment]
Following a conversation with @jni on gitter, I'm leaning towards the last possibility (``background=0`` by default).

**Comment by ahojnnes** [comment]
+1 for background=0 by default and an intuitive example in the doc
string to show the behavior.

**Comment by emmanuelle** [comment]
Thank you very much for all the comments! Luckily there seems to be a broad consensus for ``background=0`` by default.

**Comment by JDWarner** [comment]
I believe this is closed by #1956, feel free to reopen if original issue is not resolved.
